release/models/aft/openconfig-aft-common.yang

```diff
@@ -204,6 +212,33 @@ submodule openconfig-aft-common {
             description
               "State parameters relating to IP-in-IP encapsulation.";
             uses aft-common-entry-nexthop-ipip-state;
+            uses aft-common-entry-nexthop-encap-outer-header-state;
+          }
+        }
+
+        container gre {
+          description
+            "When specified, the packet has an GRE header applied to it before
+             forwarding to the specified next-hop.";
+
+          container state {
+            config false;
+            description
+              "State parameters relating to GRE encapsulation.";
+            uses aft-common-entry-nexthop-encap-outer-header-state;
+          }
+        }
+
+        container mpls {
+          description
+            "When specified, the packet has an MPLS header applied to it before
```

Comment:
If multiple labels are getting added and each label has its own tos/ttl values then what should be the values?
Should these mpls specific leaves tos/ttl kept under "pushed-mpls-label-stack"?

Reply:
@romeyod any comments on this suggestion?

Reply:
@krishna-juniper Thanks for the suggestion

`pushed-mpls-label-stack` is of type `oc-mplst:mpls-label` which is also used in other models. Example: `popped-mpls-label-stack`, `start-label-value` [more occurences](https://github.com/search?q=repo%3Aopenconfig%2Fpublic%20mpls-label&type=code)

Instead of adding ttl/tos to `oc-mplst:mpls-label` and affecting unrelated models, we can do the following:
define a `typedef encap-outer-header-state` with ttl and tos 
*and* 
in place of mpls container, under `next-hop/state` at the same level as `pushed-mpls-label-stack` we define 
```
leaf-list pushed-mpls-label-header-state {
    type encap-outer-header-state;
}
```
This new `leaf-list pushed-mpls-label-header-state` will have ordered entries with tos/ttl corresponding to `pushed-mpls-label-stack` 

Let me know what you think about this proposal.

Reply:
The approach assumes implicit correlation between labels and ttl/tos. (labels coming from pushed-mpls-label-stack, ttl/tos coming from pushed-mpls-label-header-state).
By adding label stack as part of new container(container for encap-outer-header-state + pushed labels) we can avoid this implicit correlation at the cost of redundant information.
what are your thoughts on this approach?

Reply:
Is this approach supported on any system? Please can we provide _multi-vendor_ examples if we are going to make such a change?

Note that changes around how the MPLS label stack is being pushed have implications in gRIBI and are backwards compatible.  I'd like to make sure we're supporting a use case that is actually supported here, since it also has overhead for those that don't care about this.

I also don't really like this typedef being a single leaf -- i think we rather need to think about how this is actually user friendly to parse.